internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -569,14 +570,30 @@ func (cf CloudFormation) deleteAndRenderStack(name, description string, deleteFn
 	return nil
 }
 
+type errFailedService struct {
+	stackName string
+	status    string
+}
+
+func (e *errFailedService) RecommendActions() string {
+	return fmt.Sprintf("You may fix the error by updating the service code or the manifest configuration.\n"+
+		"You can then retry deploying your service by running %s.", color.HighlightCode("copilot svc deploy"))
+}
+func (e *errFailedService) Error() string {
+	return fmt.Sprintf("stack %s did not complete successfully and exited with status %s", e.stackName, e.status)
+}
+
 func (cf CloudFormation) errOnFailedStack(stackName string) error {
 	stack, err := cf.cfnClient.Describe(stackName)
 	if err != nil {
 		return err
 	}
 	status := aws.StringValue(stack.StackStatus)
 	if cloudformation.StackStatus(status).IsFailure() {
-		return fmt.Errorf("stack %s did not complete successfully and exited with status %s", stackName, status)
+		return &errFailedService{
+			stackName: stackName,
+			status: status,
+		}
```

Comment:
I feel like we should return this error only if the AppRunner resource is the one that's failing - considering how this is part of the effort to return actionable error message for AR only. 

We can probably rely on either [DescribeStackResource](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStackResource.html) or [DescribeStackEvents](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStackResource.html) to isolate failures that are due to AppRunner.  Even better, perhaps we could considering reusing [ErrorEvents](https://github.com/aws/copilot-cli/blob/7dc2153907df656b5b29615c7e30f424a5f9f136/internal/pkg/aws/cloudformation/cloudformation.go#L328), and iterate through the error events, decide whether there is one whose `ResourceType` is `AWS::AppRunner::Service`, and return this customized error only if there is one.

Reply:
I think I would agree with this if we were currently showing errors/Recommended actions specific to App Runner. However, currently we are showing a generic error which lets the users know that they can make manifest changes. Since the Stack Failing could happen either because of 1) IAM create error and 2) App Runner create error, it makes sense to have the generic Recommended actions for both the cases as both the cases are handled with manifest updates and the recommended action is not specific to App Runner.

Reply:
TBH I'm still not super duper convinced that users do not already know that they should update manifest/application code 😂 Especially when we have already mentioned at the end of `env init`  and  `svc init` that "you can configure your manifest". 

Some other times, the stack update fails because of transient error, user's manual modification outside of Copilot, and, unfortunately, our bugs. I'm not sure if it's soothing that we still ask users to "update your configuration / application code" in these cases 🥲 

These were the main reasons why I try to avoid returning this error on _all_ stack failures.

Reply:
Understood. Everything included now.